FirebaseStorage/Sources/FIRStorageUtils.m

```diff
@@ -134,6 +134,21 @@ + (NSError *)storageErrorWithDescription:(NSString *)description code:(NSInteger
                          userInfo:@{NSLocalizedDescriptionKey : description}];
 }
 
++ (NSTimeInterval)computeRetryIntervalFromRetryTime:(NSTimeInterval)retryTime {
+  // GTMSessionFetcher's reties start at 1 second and then double every time. We use this
+  // information to compute a best-effort estimate of what to translate the user provided retry
+  // time into.
+  NSTimeInterval lastInterval = 1.0;
+  NSTimeInterval sumOfAllIntervals = 1.0;
+
+  while (retryTime < sumOfAllIntervals) {
+    lastInterval *= 2;
+    sumOfAllIntervals += lastInterval;
+  }
+
+  return lastInterval;
```

Comment:
Would it be sufficient instead of calculating this series to just return `retryTime / 2`?

Reply:
I think this is actually `2 << (log2(retryTime) - 1)`, the numbers for retryTime/2 start to diverge as the retryTime goes up:

```
retryTime 0: PR=1, suggestion=0, log=0
retryTime 1: PR=1, suggestion=0, log=0
retryTime 2: PR=2, suggestion=1, log=2
retryTime 3: PR=2, suggestion=1, log=2
retryTime 4: PR=4, suggestion=2, log=4
retryTime 5: PR=4, suggestion=2, log=4
retryTime 6: PR=4, suggestion=3, log=4
retryTime 7: PR=4, suggestion=3, log=4
retryTime 8: PR=8, suggestion=4, log=8
retryTime 9: PR=8, suggestion=4, log=8
retryTime 10: PR=8, suggestion=5, log=8
retryTime 11: PR=8, suggestion=5, log=8
retryTime 12: PR=8, suggestion=6, log=8
retryTime 13: PR=8, suggestion=6, log=8
retryTime 14: PR=8, suggestion=7, log=8
retryTime 15: PR=8, suggestion=7, log=8
retryTime 16: PR=16, suggestion=8, log=16
retryTime 17: PR=16, suggestion=8, log=16
retryTime 18: PR=16, suggestion=9, log=16
retryTime 19: PR=16, suggestion=9, log=16
retryTime 20: PR=16, suggestion=10, log=16
retryTime 21: PR=16, suggestion=10, log=16
retryTime 22: PR=16, suggestion=11, log=16
retryTime 23: PR=16, suggestion=11, log=16
retryTime 24: PR=16, suggestion=12, log=16
retryTime 25: PR=16, suggestion=12, log=16
retryTime 26: PR=16, suggestion=13, log=16
retryTime 27: PR=16, suggestion=13, log=16
retryTime 28: PR=16, suggestion=14, log=16
retryTime 29: PR=16, suggestion=14, log=16
retryTime 30: PR=16, suggestion=15, log=16
retryTime 31: PR=16, suggestion=15, log=16
retryTime 32: PR=32, suggestion=16, log=32
retryTime 33: PR=32, suggestion=16, log=32
retryTime 34: PR=32, suggestion=17, log=32
retryTime 35: PR=32, suggestion=17, log=32
retryTime 36: PR=32, suggestion=18, log=32
retryTime 37: PR=32, suggestion=18, log=32
retryTime 38: PR=32, suggestion=19, log=32
retryTime 39: PR=32, suggestion=19, log=32
retryTime 40: PR=32, suggestion=20, log=32
retryTime 41: PR=32, suggestion=20, log=32
retryTime 42: PR=32, suggestion=21, log=32
retryTime 43: PR=32, suggestion=21, log=32
retryTime 44: PR=32, suggestion=22, log=32
retryTime 45: PR=32, suggestion=22, log=32
retryTime 46: PR=32, suggestion=23, log=32
retryTime 47: PR=32, suggestion=23, log=32
retryTime 48: PR=32, suggestion=24, log=32
retryTime 49: PR=32, suggestion=24, log=32
retryTime 50: PR=32, suggestion=25, log=32
retryTime 51: PR=32, suggestion=25, log=32
retryTime 52: PR=32, suggestion=26, log=32
retryTime 53: PR=32, suggestion=26, log=32
retryTime 54: PR=32, suggestion=27, log=32
retryTime 55: PR=32, suggestion=27, log=32
retryTime 56: PR=32, suggestion=28, log=32
retryTime 57: PR=32, suggestion=28, log=32
retryTime 58: PR=32, suggestion=29, log=32
retryTime 59: PR=32, suggestion=29, log=32
retryTime 60: PR=32, suggestion=30, log=32
retryTime 61: PR=32, suggestion=30, log=32
retryTime 62: PR=32, suggestion=31, log=32
retryTime 63: PR=32, suggestion=31, log=32
retryTime 64: PR=64, suggestion=32, log=64
retryTime 65: PR=64, suggestion=32, log=64
retryTime 66: PR=64, suggestion=33, log=64
retryTime 67: PR=64, suggestion=33, log=64
retryTime 68: PR=64, suggestion=34, log=64
retryTime 69: PR=64, suggestion=34, log=64
retryTime 70: PR=64, suggestion=35, log=64
retryTime 71: PR=64, suggestion=35, log=64
retryTime 72: PR=64, suggestion=36, log=64
retryTime 73: PR=64, suggestion=36, log=64
retryTime 74: PR=64, suggestion=37, log=64
retryTime 75: PR=64, suggestion=37, log=64
retryTime 76: PR=64, suggestion=38, log=64
retryTime 77: PR=64, suggestion=38, log=64
retryTime 78: PR=64, suggestion=39, log=64
retryTime 79: PR=64, suggestion=39, log=64
retryTime 80: PR=64, suggestion=40, log=64
retryTime 81: PR=64, suggestion=40, log=64
retryTime 82: PR=64, suggestion=41, log=64
retryTime 83: PR=64, suggestion=41, log=64
retryTime 84: PR=64, suggestion=42, log=64
retryTime 85: PR=64, suggestion=42, log=64
retryTime 86: PR=64, suggestion=43, log=64
retryTime 87: PR=64, suggestion=43, log=64
retryTime 88: PR=64, suggestion=44, log=64
retryTime 89: PR=64, suggestion=44, log=64
retryTime 90: PR=64, suggestion=45, log=64
retryTime 91: PR=64, suggestion=45, log=64
retryTime 92: PR=64, suggestion=46, log=64
retryTime 93: PR=64, suggestion=46, log=64
retryTime 94: PR=64, suggestion=47, log=64
retryTime 95: PR=64, suggestion=47, log=64
retryTime 96: PR=64, suggestion=48, log=64
retryTime 97: PR=64, suggestion=48, log=64
retryTime 98: PR=64, suggestion=49, log=64
retryTime 99: PR=64, suggestion=49, log=64
```